python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -608,6 +609,157 @@ def _align_to_index(
+        """
+        Construct a GeoSeries of MULTIPOINTs from an array of interleaved xy
+        coordinates.
+
+        Parameters
+        ----------
+        points_xy: array-like
+            Coordinates of the points, interpreted as interlaved x-y coords.
+        geometry_offset: array-like
+            Offsets indicating the starting index of the multipoint. Multiply
+            the index by 2 results in the starting index of the coordinate.
+            See example for detail.
+
+        Returns
+        -------
+        GeoSeries:
+            A GeoSeries made of the points.
+
+        Example
+        -------
+        >>> import numpy as np
+        >>> cuspatial.GeoSeries.from_multipoints_xy(
+        ...     np.array([0, 0, 1, 1, 2, 2, 3, 3], dtype='f8'),
+        ...     np.array([0, 2, 4], dtype='i4'))
+        0    MULTIPOINT (0.00000 0.00000, 1.00000 1.00000)
+        1    MULTIPOINT (2.00000 2.00000, 3.00000 3.00000)
+        dtype: geometry
+        """
+        return cls(
+            GeoColumn._from_multipoints_xy(
+                as_column(multipoints_xy), as_column(geometry_offset)
+            )
+        )
+
+    @classmethod
+    def from_linestrings_xy(
+        cls, linestrings_xy, geometry_offset, part_offset
+    ) -> T:
+        """
+        Construct a GeoSeries of LINESTRINGs from an array of interleaved xy
+        coordinates.
+
+        Parameters
+        ----------
+        linestrings_xy : array-like
+            Coordinates of the points, interpreted as interleaved x-y coords.
+        geometry_offset : array-like
+            Offsets of the first coordinate of each geometry. The length of
+            this array is the number of geometries.  Offsets with a difference
+            greater than 1 indicate a MultiLinestring.
+        part_offset : array-like
+            Offsets into the coordinates array indicating the beginning of
+            each part. The length of this array is the number of parts.
+
+        Returns
+        -------
+        GeoSeries:
+            A GeoSeries of LINESTRINGs.
+
+        Example
+        -------
+        >>> import cudf
+        >>> import cuspatial
+        >>> linestrings_xy = cudf.Series(
+                [0.0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5])
+        >>> geometry_offset = cudf.Series([0, 1])
+        >>> part_offset = cudf.Series([0, 6])
+        >>> cuspatial.GeoSeries.from_linestrings_xy(
+                linestrings_xy, geometry_offset, part_offset)
+        0    LINESTRING (0 0, 1 1, 2 2, 3 3, 4 4, 5 5)
+        dtype: geometry
+        """
+        return cls(
+            GeoColumn._from_linestrings_xy(
+                as_column(linestrings_xy),
+                as_column(geometry_offset, dtype="int32"),
+                as_column(part_offset, dtype="int32"),
+            )
+        )
+
+    @classmethod
+    def from_polygons_xy(
+        cls, polygons_xy, geometry_offset, part_offset, ring_offset
```

Comment:
Besides, it may be my personal thing. I kinda like ordering from `xy->ring->part->geometry` as it follows the bottom-up traversal order.

Reply:
Agreed, I can go with that.

Reply:
Done.